### PR TITLE
feat: Add a confirmation screen

### PR DESCRIPTION
**Pull request by jorelosorio**
Closes https://github.com/AtB-AS/kundevendt/issues/15878

### CODE REVIEW

**Review by marius-at-atb — APPROVED** [review]
Nice and clean! 🧹 

Something to think about when using this, is the transition in and out. I'm guessing a fade transition would be good.

**Comment by jorelosorio** [comment]
> Nice and clean! 🧹
> 
> Something to think about when using this, is the transition in and out. I'm guessing a fade transition would be good.

For now, I think it takes the default one.

File: src/stacks-hierarchy/Root_ConfirmationScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      >
+        {message}
+      </ThemeText>
+      <View
+        style={[
+          styles.circle,
+          {backgroundColor: circleColor.background},
+        ]}
+      >
+        <ThemeIcon size="large" svg={ConfirmSvg} fill={themeColor.text} />
```
Comment (gorandalum) [inline src/stacks-hierarchy/Root_ConfirmationScreen.tsx:47]: Should not use `fill` parameter, that is for overriding with colors outside of the design system. Try using `colorType` instead.

**Review by gorandalum — CHANGES_REQUESTED** [review]
GW! Looks pretty clean :+1: Some comments on some subtle improvements that can be made, but the one regarding function as screen params should at least be looked on 😊

File: src/stacks-hierarchy/navigation-types.ts [hunk 2]
```diff
@@ -122,6 +122,13 @@ type Root_PurchasePaymentWithCreditCardScreenParams = PaymentParams & {
   paymentMethod: CardPaymentMethod;
 };
 
+type Root_ConfirmationScreenParams = {
+  message: string;
+  // Time that must be wait until onComplete is called (in milliseconds)
+  delayBeforeCompleted?: number;
+  onCompleted: () => void;
```
[the remark this answers is not in the document]
Comment (jorelosorio) [reply src/stacks-hierarchy/navigation-types.ts]: Fixed.

File: src/stacks-hierarchy/Root_ConfirmationScreen.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorelosorio) [reply src/stacks-hierarchy/Root_ConfirmationScreen.tsx]: Fixed

File: src/utils/use-complete-onboarding-and-enter-app.ts [hunk 3]
```diff
@@ -20,14 +20,13 @@ export const useCompleteOnboardingAndEnterApp = () => {
...
-     */
-  params: RootStackParamList[T];
-};
+export type NextScreenParams<T extends keyof RootStackParamList = keyof RootStackParamList> = {
+  [S in T]: {
+    screen: S;
+    params: RootStackParamList[S];
+  };
+}[T];
+
```
Comment (gorandalum) [inline src/utils/use-complete-onboarding-and-enter-app.ts:29]: Could we move this to stacks-hierarchy/navigation-types? It is now used in a non-onboarding-context.

File: src/stacks-hierarchy/navigation-types.ts [hunk 4]
```diff
@@ -112,6 +112,13 @@ type Root_PurchasePaymentWithCreditCardScreenParams = PaymentParams & {
   paymentMethod: CardPaymentMethod;
 };
 
+type Root_ConfirmationScreenParams = {
+  message: string;
+  // Time that must be wait until onComplete is called (in milliseconds)
+  delayBeforeCompleted?: number;
+  nextScreen: NextScreenParams<'Root_TabNavigatorStack'>;
```
Comment (gorandalum) [inline src/stacks-hierarchy/navigation-types.ts:119]: Do you need the generics here anymore with the updated typing for `NextScreenParams`?

**Review by gorandalum — APPROVED** [review]

File: src/stacks-hierarchy/Root_ConfirmationScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  route: {
+    params: {message, delayBeforeCompleted, nextScreen},
+  },
+}: Props) => {
+  const styles = useStyles();
+  const {theme} = useTheme();
+
+  useEffect(() => {
+    const timer = setTimeout(
+      () => navigation.navigate(nextScreen.screen, nextScreen.params),
```
Comment (jorelosorio) [inline src/stacks-hierarchy/Root_ConfirmationScreen.tsx:28]: By removing the generic as mentioned here: https://github.com/AtB-AS/mittatb-app/pull/4237/files#r1461808581 the compiler complains:

image

Comment (jorelosorio) [reply src/stacks-hierarchy/Root_ConfirmationScreen.tsx:28]: I leave the generic set in order to avoid this issue, lets have a look when implementing.

**Pull request merged**